commit 5a3a6f90a5f34b17c2af307932f39fa80958549b
Author: Igor <igorwwwwwwwwwwwwwwwwwwww@users.noreply.github.com>
Date:   Thu Mar 25 15:16:03 2021 +0100

    Backport (6.0): Introduce fast path to bypass expensive serveClientsBlockedOnKeyByModule call
    
    This is a backport of #8689 to the 6.0 branch.

diff --git a/src/blocked.c b/src/blocked.c
index 396a71569..460362ceb 100644
--- a/src/blocked.c
+++ b/src/blocked.c
@@ -425,6 +425,10 @@ void serveClientsBlockedOnStreamKey(robj *o, readyList *rl) {
 void serveClientsBlockedOnKeyByModule(readyList *rl) {
     dictEntry *de;
 
+    /* Optimization: If no clients are in type BLOCKED_MODULE,
+     * we can skip this loop. */
+    if (!server.blocked_clients_by_type[BLOCKED_MODULE]) return;
+
     /* We serve clients in the same order they blocked for
      * this key, from the first blocked to the last. */
     de = dictFind(rl->db->blocking_keys,rl->key);
